pkg/apis/edge/v1alpha1/edge-placement.go

```diff
@@ -107,10 +104,16 @@ type NonNamespacedObjectReferenceSet struct {
 	// Empty list means nothing matches.
 	Resources []string `json:"resources"`
 
-	// `resourceNames` is a list of objects that match by name.
+	// `namespaceSelectors` tests the labels on the object's namespace.
+	// A non-namespaced object matches if and only if this list is empty.
+	// A namespaced object matches if and only if the namespace's labels match
+	// at least one member of this list.
+	NamespaceSelectors []metav1.LabelSelector `json:"namespaceSelectors"`
+
+	// `objectNames` is a list of object names that match.
 	// An entry of `"*"` means that all match.
 	// Empty list means nothing matches.
-	ResourceNames []string `json:"resourceNames,omitempty"`
+	ObjectNames []string `json:"objectNames,omitempty"`
 
 	// `labelSelectors` allows matching objects by a rule rather than listing individuals.
```

Comment:
Does all the labels represent a single predicate that matches all the labels? i.e., logical AND
or a subset? i.e., logical OR ?
Is it possible to have more than one predicate if all labels needs to match?

Reply:
I added a clarification here.